File: update_icons.py
import os
from PIL import Image

def resize_icons():
    # Source icon path
    source_path = r"C:\Users\RAMLA\.gemini\antigravity\brain\347f346c-ead4-46de-8822-36e9ce7f7260\modern_music_icon_1767180012228.png"
    
    # Destination base path
    android_res_path = r"C:\Users\RAMLA\Downloads\mtc-player\android\app\src\main\res"
    
    # Standard Android Icon Sizes
    sizes = {
        "mipmap-mdpi": (48, 48),
        "mipmap-hdpi": (72, 72),
        "mipmap-xhdpi": (96, 96),
        "mipmap-xxhdpi": (144, 144),
        "mipmap-xxxhdpi": (192, 192)
    }

    if not os.path.exists(source_path):
        print(f"Error: Source file not found at {source_path}")
        return

    try:
        img = Image.open(source_path)
        print(f"Opened source image: {source_path}")

        for folder, size in sizes.items():
            dest_folder = os.path.join(android_res_path, folder)
            os.makedirs(dest_folder, exist_ok=True)
            
            # Resize
            resized_img = img.resize(size, Image.Resampling.LANCZOS)
            
            # Save as ic_launcher.png
            icon_path = os.path.join(dest_folder, "ic_launcher.png")
            resized_img.save(icon_path)
            print(f"Saved {folder}/ic_launcher.png ({size})")
            
            # Save as ic_launcher_round.png (using same image for simplicity, usually masked circle)
            round_icon_path = os.path.join(dest_folder, "ic_launcher_round.png")
            resized_img.save(round_icon_path)
            print(f"Saved {folder}/ic_launcher_round.png ({size})")
            
        print("All icons updated successfully.")
        
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    resize_icons()

File: update_icons.js
const Jimp = require('jimp');
const path = require('path');
const fs = require('fs');

async function resizeIcons() {
    // Use raw strings or double backslashes for paths
    const sourcePath = path.join(__dirname, 'resources', 'icon.png');
    const androidResPath = path.join(__dirname, 'android', 'app', 'src', 'main', 'res');
    const pwaIconPath = path.join(__dirname, 'public', 'pwa-icon.png');

    const sizes = {
        "mipmap-mdpi": 48,
        "mipmap-hdpi": 72,
        "mipmap-xhdpi": 96,
        "mipmap-xxhdpi": 144,
        "mipmap-xxxhdpi": 192
    };

    if (!fs.existsSync(sourcePath)) {
        console.error(`Error: Source file not found at ${sourcePath}`);
        return;
    }

    try {
        console.log("Reading source image...");
        const image = await Jimp.read(sourcePath);
        console.log(`Opened source image: ${sourcePath}`);

        for (const [folder, size] of Object.entries(sizes)) {
            const destFolder = path.join(androidResPath, folder);
            if (!fs.existsSync(destFolder)) fs.mkdirSync(destFolder, { recursive: true });

            const iconPath = path.join(destFolder, "ic_launcher.png");
            const roundIconPath = path.join(destFolder, "ic_launcher_round.png");

            await image.clone().resize(size, size).writeAsync(iconPath);
            console.log(`Saved ${folder}/ic_launcher.png (${size}x${size})`);

            // For round icon, we can try to apply a circle mask if we want to be fancy, 
            // but just resizing is often enough if the icon is designed well. 
            // However, Android expects round icons to be circular.
            // Jimp has a circle() method.
            await image.clone().resize(size, size).circle().writeAsync(roundIconPath);
            console.log(`Saved ${folder}/ic_launcher_round.png (${size}x${size})`);
        }

        // Generate PWA Icon
        await image.clone().resize(512, 512).writeAsync(pwaIconPath);
        console.log(`Saved public/pwa-icon.png (512x512)`);
        console.log("All icons updated successfully.");
    } catch (error) {
        console.error("An error occurred:", error);
        process.exit(1);
    }
}

resizeIcons();
